web/packages/teleterm/src/ui/Search/pickers/ActionPicker.tsx

```diff
@@ -873,22 +868,55 @@ function HighlightField(props: {
 
 function FilterButton(props: { text: string; onClick(): void }) {
   return (
-    <ButtonPrimary
-      px={2}
+    <Flex
+      justifyContent="center"
+      alignItems="center"
+      css={`
+        color: ${props => props.theme.colors.buttons.text};
+        background: ${props => props.theme.colors.spotBackground[1]};
+        border-radius: ${props => props.theme.radii[2]}px;
+      `}
+      px="6px"
       size="small"
-      title={props.text}
-      onClick={props.onClick}
     >
+      <CloseIcon
+        color="buttons.text"
+        mr={1}
+        mt="1px"
+        title="Remove filter"
+        onClick={props.onClick}
+        css={`
+          cursor: pointer;
+          border-radius: ${props => props.theme.radii[1]}px;
+
+          :hover {
+            background: ${props => props.theme.colors.spotBackground[1]};
+          }
+
+          > svg {
+            height: 13px;
+            width: 13px;
+          }
+        `}
+      />
       <span
+        title={props.text}
         css={`
           max-width: calc(${props => props.theme.space[9]}px * 2);
```

Comment:
I think we can remove max-width from here, with unified resources and the feedback we got about the connection picker we know how users feel about identifiers being cut off.

Without max-width, very long cluster names will be "naturally" cut off.

![long cluster name](https://github.com/gravitational/teleport/assets/27113/f6c50ae3-c519-4dbd-9c19-1d9d38d408f7)

Reply:
Done, I also removed `text-overflow` and `overflow`, they don't have any effect without `max-width`.